### PR TITLE
test: add pytest for fsread1 max_read_size validation

### PR DESCRIPTION
**Pull request by jelly**

### CODE REVIEW

File: test/pytest/mocktransport.py [hunk 1]
```diff
@@ -13,7 +13,8 @@ class MockTransport(asyncio.Transport):
     close_future: Optional[asyncio.Future] = None
 
     def send_json(self, _channel: str, **kwargs) -> None:
-        msg = {k.replace('_', '-'): v for k, v in kwargs.items()}
+        # max_read_size is one of our special keys which uses underscores
+        msg = {k.replace('_', '-') if k != "max_read_size" else k: v for k, v in kwargs.items()}
```
Comment (allisonkarlitskaya) [inline test/pytest/mocktransport.py:17]: I'd prefer to avoid the hack here.  Can you just use `.send_data()` from the call site?

**Review by allisonkarlitskaya — CHANGES_REQUESTED** [review]

File: test/pytest/mocktransport.py (hunk 1 above)
Comment (jelly) [reply test/pytest/mocktransport.py]: It is a bit more elaborate but I can make it happen :)

**Review by allisonkarlitskaya — DISMISSED** [review]
thx

**Comment by allisonkarlitskaya** [comment]
Note: unit test failures are unrelated.  We can rebase after #19208 if we want to be extra careful.

**Review by martinpitt — APPROVED** [review]
Cheers!

**Pull request merged**